aslam_backend_expressions/include/aslam/backend/implementation/VectorExpressionNode.hpp

```diff
@@ -27,6 +27,40 @@ void ConstantVectorExpressionNode<D>::accept(ExpressionNodeVisitor& visitor) {
   visitor.visit("#", this);
 }
 
+template <int D>
+void StackedScalarVectorExpressionNode<D>::accept(ExpressionNodeVisitor& visitor) {
+  //New line or what do we want here? (for sure would want to expand)
+  visitor.visit("#", this, components.at(0));
```

Comment:
Do you know of some reasonable easy way to expand an std array to variadic template args (like pythons *)
The only thing I found was some rather complicated helper structs doing template magic

Reply:
I dont really understand the visitor either (probably less than you do), but from looking at the examples it looks to me like it would try to print the expression (i.e. the AdditionNode looks like ("+", this, lhs, rhs)
But also I'm not even sure how it should look for a stacking
Any ideas/opinions with this (slightly more) information @HannesSommer ?

Reply:
I finally found some time to look into this. I think you are right. It is compute a text representing the expression tree as well as its value.

No I also understand you question. And I agree, you'd like to pass all components at once as arguments. 
Thus, I think could use `std::apply` when prepending the missing arguments first (e.g. using `tuple_cat`) -- if visit wasn't a template -- but of course it is a template function.
If I read the code right, there might be a more direct way that also works with the template function that `visit` is. 

Did you try to pass the std::array directly as a single argument after `this` after converting it to a tuple? E.g.

```suggestion
  visitor.visit("#", this, std::tuple_cat(components));
```

I could imagine that this propagates through and end up selecting the copy constructor of the internal `args`-tuple of the [NodeImpl struct](https://github.com/ethz-asl/aslam_optimizer/commit/85e9ae8720e0416cd4c55572ea04e6b48e221304#diff-b975090fb63fca6278d226b2d58540ccaef9c28d16a8654b9488cdd2c975a7efR128).

I did some encouraging experiments regarding this [here](https://godbolt.org/z/zjf8cfK95). I'm keeping my :crossed_fingers: !

Reply:
I dont think that works on its own, because what happens in your compiler explorer example is that the copy constructor of tuple is called.
But I did not know tuple_cat and it seemed useful, so what seems to work is using std::apply as you suggested (even it being a template), but it requires c++ 17 :/
see: https://godbolt.org/z/xnar5nf68